runtime/lua/vim/lsp/util.lua

```diff
@@ -423,27 +423,44 @@ function M.make_floating_popup_options(width, height, opts)
   }
 end
 
+local function should_add_item(item)
+  local stack = vim.fn.gettagstack()
+  -- Check if we're at the bottom of the tagstack.
+  if stack.curidx <= 1 then return true end
+
+  local top_item = stack.items[stack.curidx-1]
+  -- Check if the item at the top of the tagstack is exactly the
+  -- same we want to push.
+  if top_item.tagname ~= item.tagname then return true end
+  for i, v in ipairs(top_item.from) do
+    if v ~= item.from[i] then return true end
+  end
+  return false
+end
+
 function M.jump_to_location(location)
   -- location may be Location or LocationLink
   local uri = location.uri or location.targetUri
   if uri == nil then return end
-  local bufnr = vim.uri_to_bufnr(uri)
-  -- Save position in jumplist
-  vim.cmd "normal! m'"
 
-  -- Push a new item into tagstack
   local from = {vim.fn.bufnr('%'), vim.fn.line('.'), vim.fn.col('.'), 0}
-  local items = {{tagname=vim.fn.expand('<cword>'), from=from}}
-  vim.fn.settagstack(vim.fn.win_getid(), {items=items}, 't')
-
-  --- Jump to new location
-  api.nvim_set_current_buf(bufnr)
-  api.nvim_buf_set_option(0, 'buflisted', true)
+  local item = {tagname=vim.fn.expand('<cword>'), from=from}
   local range = location.range or location.targetSelectionRange
   local row = range.start.line
   local col = range.start.character
-  local line = api.nvim_buf_get_lines(0, row, row+1, true)[1]
-  col = vim.str_byteindex(line, col)
```

Comment:
This removes a correction that is necessary to get correct column information when the line contains non-ASCII characters before `col` (UTF-16 encoding...)

Reply:
Oh, I see. Thanks for pointing that out. I'll revert this.